pkg/endpoint/bpf.go

```diff
@@ -542,26 +545,63 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir, reason string) (uint64, err
 		}
 	}
 
+	var (
+		consumersAdd, consumersRm policy.SecurityIDContexts
+		policyChanged             bool
+	)
 	// Only generate & populate policy map if a seclabel and consumer model is set up
 	if c != nil {
 		c.AddMap(e.PolicyMap)
 
 		// Regenerate policy and apply any options resulting in the
 		// policy change.
 		// This also populates e.PolicyMap
-		var (
-			consumersToRm, consumersAdd policy.RuleContexts
-			flushEndpointCT             bool
-		)
-		_, flushEndpointCT, consumersAdd, consumersToRm, err = e.regeneratePolicy(owner, nil)
+		policyChanged, consumersAdd, consumersRm, err = e.regeneratePolicy(owner, nil)
 		if err != nil {
 			e.Mutex.Unlock()
-			err = fmt.Errorf("Unable to regenerate policy for '%s': %s",
-				e.PolicyMap.String(), err)
-			return 0, err
+			return 0, fmt.Errorf("Unable to regenerate policy for '%s': %s", e.PolicyMap.String(), err)
+		}
+		// policyChanged can still be true and, at the same time,
```

Comment:
Is it possible to only have L4 rules i.e no L7 rules or L3 rules, in which case, there will be no proxy redirect port to forward to. Will that scenario be handled correctly here?